src/event/log_event.rs

```diff
@@ -12,43 +14,71 @@ use std::{
     iter::FromIterator,
 };
 
-#[derive(Clone, Debug, Default, Getters, PartialEq)]
+#[derive(Clone, Debug, Getters, PartialEq, Derivative, Deserialize)]
 pub struct LogEvent {
-    fields: BTreeMap<String, Value>,
+    // **IMPORTANT:** Due to numerous legacy reasons this **must** be a Map variant.
+    #[serde(flatten)]
```

Comment:
Good spot.

Indeed, this test does fail because it attempts to deserialize a string into metadata.

```rust
    #[test]
    fn serialize_metadata_field() {
        let mut event = LogEvent::default();
        event.insert("message", "spork");
        event.insert("metadata", "zork");

        let serialized = serde_json::to_string(&event).unwrap();
        let deserialized: LogEvent = serde_json::from_str(&serialized).unwrap();

        assert_eq!(event, deserialized);
    }
```

Do you know what should happen here? It doesn't look like Value is symettric.. I can write a custom Deserializer, but that still won't get the above test to pass because of Value.

Reply:
In the short term, it looks like the easy solution is to just skip serializing and deserializing the metadata. This could be done with a `#[serde(skip)]` on the `metadata` field. I will have to deal with the metadata separately in my changes.